### PR TITLE
[eclipse/xtext#1837] converted test code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1837] converted test code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		@Override
+		public String getIndentString() {
+			return "  ";
+		}
+	}
+
+	@Override
+	protected com.google.inject.Module getRuntimeModule() {
+		return new StatemachineRuntimeModule() {
+			@SuppressWarnings("unused")
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:34]: move above method

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			public Class<? extends IIndentationInformation> bindIIndentationInformation() {
+				return Bug489571Test.Bug489571IndentationInformation.class;
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:36]: Remove "Bug489571Test."

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			}
+
+			@SuppressWarnings("unused")
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:39]: remove when on method level

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			public Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
+				return AbstractWebServerTest.TestLineSeparatorInformation.class;
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:41]: remove "AbstractWebServerTest."

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.java [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+			return "\n";
+		}
+	}
+
+	@Override
+	protected com.google.inject.Module getIdeModule() {
+		return new StatemachineIdeModule() {
+			@SuppressWarnings("unused")
+			public Class<? extends StatemachineTemplateProposalProvider> bindStatemachineTemplateProposalProvider() {
+				return ContentAssistTest.StatemachineTemplateProposalProvider2.class;
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.java:39]: remove "ContentAssistTest."

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.java [hunk 6]
```diff
@@ -0,0 +1,538 @@
...
+			}
+		};
+	}
+
+	protected void assertContentAssistResult(CharSequence resourceContent, CharSequence expectedResult) {
+		String contentString = resourceContent.toString();
+		int cursorOffset = contentString.indexOf("|");
+		if (cursorOffset >= 0) {
+			contentString = contentString.substring(0, cursorOffset)+ contentString.substring(cursorOffset+ 1);
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.java:48]: spaces before "+"

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/GeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+				boolean hasElements = false;
+				for (State state : statemachine.getStates()) {
+					if (!hasElements) {
+						hasElements = true;
+					} else {
+						builder.appendImmediate(",", "");
+					}
+					String _name = state.getName();
+					builder.append(_name);
+				}
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/GeneratorTest.java:54]: Replace by
```
builder.append(statemachine.getStates().stream().map(State::getName).collect(Collectors.joining(",")));
```

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/UpdateDocumentTest.java [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+	@Singleton
+	public static class TestResourceValidator extends ResourceValidatorImpl {
+		private Thread workerThread;
+
+		private long sleepTime;
+
+		private volatile boolean canceled;
+
+		private int entryCounter;
+
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/UpdateDocumentTest.java:53]: remove empty lines between fields

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/UpdateDocumentTest.java [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+			return entryCounter;
+		}
+
+		public int getExitCounter() {
+			return exitCounter;
+		}
+	}
+
+	@Inject
+	private UpdateDocumentTest.TestResourceValidator resourceValidator;
```
Comment (kthoms) [inline org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/UpdateDocumentTest.java:128]: Remove "UpdateDocumentTest."

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:34]: it is above  the unused method ?!?

File: org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.java:39]: same as above

**Review by szarnekow — APPROVED** [review]

File: org.eclipse.xtext.web.example.statemachine.ide/src/org/eclipse/xtext/web/example/statemachine/ide/StatemachineWebContentProposalProvider.java [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+	private StatemachineGrammarAccess statemachineGrammarAccess;
+
+	@Inject
+	private StatemachineTemplateProposalProvider templateProvider;
+
+	@Override
+	protected void _createProposals(RuleCall ruleCall, ContentAssistContext context,
+			IIdeContentProposalAcceptor acceptor) {
+		if (Objects.equal(ruleCall.getRule(), statemachineGrammarAccess.getBOOLEANRule())) {
+			if ("true".startsWith(context.getPrefix())) {
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.web.example.statemachine.ide/src/org/eclipse/xtext/web/example/statemachine/ide/StatemachineWebContentProposalProvider.java:39]: maybe there is no prefix filtering in web?

[the remark this answers is not in the document]
Comment (szarnekow) [reply org.eclipse.xtext.web.example.statemachine.ide/src/org/eclipse/xtext/web/example/statemachine/ide/StatemachineWebContentProposalProvider.java:38]: Could well be. I'd leave the code as is.

**Pull request merged**